File: abc/abc026/b.js
'use strict';
process.stdin.setEncoding("utf8");

var lines = []; 
var reader = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

reader.on("line", (line) => {
  lines.push(line);
});
reader.on("close", () => {
    //ここに処理を記述
    let n = parseInt(lines[0], 10);
    let A = new Array();
    for (let i = 0; i < n; i++) {
        A.push(parseInt(lines[i+1]));
    }
    A.sort((a, b) => b - a);
    let ans = 0;
    for (let i = 0; i < n; i++) {
        if (i%2 === 0) {
            ans += A[i]*A[i];
        } else {
            ans -= A[i]*A[i];
        }
    }
    console.log(ans*Math.PI)
});

File: abc/abc026/b.py
import math
n = int(input())
r = [int(input()) for _ in range(n)]
r.sort(reverse=True)
ans = 0
for i in range(n):
    if i%2 == 0:
        ans += r[i]**2
    else:
        ans -= r[i]**2
print(ans*math.pi)